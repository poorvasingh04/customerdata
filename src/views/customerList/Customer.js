import { func } from 'prop-types';
import React from 'react';
import {
  View,
  Text,
} from 'react-native';
import style from './style';

const {
  rowStyle,
  textStyle,
} = style;

function Customer({
  data,
}) {

  const { user_id, name } = data;

  return (
    <View style={rowStyle}>
      <Text style={textStyle}>
        {user_id}
      </Text>
      <Text style={textStyle}>
        { name }
      </Text>
    </View>
  );
}

export default Customer;