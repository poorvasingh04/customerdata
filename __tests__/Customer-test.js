import React from 'react';
import {
  View, Text,
} from 'react-native';
import renderer from 'react-test-renderer';

import Customer from '../src/views/customerList/Customer';

 
describe('--Test Customer component--', () => {
  let testInstance;
  let data;

  beforeEach(() => {
    data = {
      user_id: '123',
      name: 'Joe',
    };
    testInstance = renderer.create(<Customer data={data} />).root;
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Customer data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders expected number of components', () => {
    expect(testInstance.findByType(View)).toBeDefined();
    expect(testInstance.findAllByType(Text).length).toBe(2);
  });

});