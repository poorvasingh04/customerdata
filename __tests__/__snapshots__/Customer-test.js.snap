// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--Test Customer component-- renders correctly 1`] = `
<View
  style={
    Object {
      "borderWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "flexGrow": 1,
        "fontSize": 16,
        "margin": 5,
        "width": 335,
      }
    }
  >
    123
  </Text>
  <Text
    style={
      Object {
        "flexGrow": 1,
        "fontSize": 16,
        "margin": 5,
        "width": 335,
      }
    }
  >
    Joe
  </Text>
</View>
`;
